docs/basic/getting-started/function-components.md

```diff
@@ -39,6 +39,30 @@ const Title: React.FunctionComponent<{ title: string }> = ({
 }) => <div title={title}>{children}</div>;
 ```
 
+<details>
+<summary>
+
+As of [@types/react PR #46643](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46643), you can use a new `React.VoidFunctionComponent` or `React.VFC` type if you wish to declare a component that explicitly does not have children.
```

Comment:
```suggestion
As of [@types/react PR #46643](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46643), you can use a new `React.VoidFunctionComponent` or `React.VFC` type if you wish to declare the accepted `children` explicitly.
```

`VoidFunctionComponent` can act as a migration path (for a planned breaking change) where you replace all `FunctionComponent` with `VoidFunctionComponent` and see what breaks. Might add that as well and mention that it's temporary as Ferdy said.